apertium-uzb.uzb.lexc

```diff
@@ -124,29 +161,72 @@ Multichar_Symbols
 
 %<guess%>     ! 
 
-%<sg%>
-%<pl%>
+! Number
+%<sg%>        ! Singular ! Birlik
+%<pl%>        ! Plural   ! Ko'plik
 
-%<prn%>
-%<adv%>
-! %<ref%>
-%<pers%>
-! %<dem%>
+! Pronoun types
+%<pers%>      ! Personal
+%<recip%>     ! Reciprocal
 
 !! Pronoun&Determiner types
 %<dem%>       ! Demonstrative
 %<ind%>       ! Indefinite
 %<itg%>       ! Interrogative
 %<qnt%>       ! Quantifier
-%<neg%>       ! Negative       !# ешкім
-              ! (NOTE: also used to denote negation in verbs, i.e for м{A})
+%<neg%>       ! Negative    !# ешкім ! (NOTE: also used to denote negation in verbs, i.e for м{A})
 %<ref%>       ! Reflexive
 
 ! Numeral types
 %<ord%>       ! Ordinal
 %<coll%>      ! Collective
 %<dist%>      ! Distributive
 
+!! Non-Finite verb forms
+
+!!! Participles
+!TODO 
+!This part was derived from Kazakh transucers and need to be applied for Uzbek as welll
+!%<prc_perf%>  ! Perfect participle
+              ! -{I}п
+              !# "Бірақ мысығы үйде, _ұйықтап_ жатыр.";
+!%<prc_impf%>  ! Imperfect participle
+              ! -{E}
+              !# "...олар далада _ойнай_ алмады...";
+!%<prc_vol%>   ! Volition participle
+              ! -{G}{I}
+              !# барғым келмейді;
+!%<prc_cond%>  ! Conditional participle
+              ! -с{A}
+              !# ...жесең болады...;
+!%<prc_fplan%> ! Future plan participle
+              ! -{M}{A}{K},
+              ! -{M}{A}{K}ш{I} Dir/LR
+              !# "Шал қазға бармақшы болады.";
+!%<prc_plan%>  ! Plan participle
+              ! -{G}{A}л{I}
+              !# мен сөйлескелі келдім;
+!!%<prc_irre%> ! Irrealis participle
+              ! -{E}{T}{I}н
+              ! (This form is analyzed as <gpr_impf> and <ger_impf>)
+              ! (FIXME this might be wrong)
+
+!!! Verbal adverbs        ! Ravishdosh        !Глагольные наречия
+%<gna_perf%>  ! -{I}b,   ! # O'sha vaqt endi quyosh _botib_, yulduzlar ko'rina boshlagan edi
+              !# "...ул вакытта инде кояш _баеп_, йолдызлар күренә башлаган
+              !# иде..." (Ф.Хөсни);
+%<gna_impf%>  ! -{A}	      
+%<gna_cond%>  ! -с{A}
+              !# ...қайда екенін _білсе_, маған бұл туралы айтыр еді...;
+%<gna_until%> ! -{G}{A}нш{A}
+              !# "Мен кеткенше ол жауап бермейді.";
+%<gna_after%> ! -{G}{A}л{I}
+              ! (NOTE: ambiguous with prc_plan)
+              !# "Сабақ _біткелі_ екі сағат өтті."; Ол кеткелі біз жұмыс
+              !# істемейдік;
```

Comment:
We need to determine the tags needed for Uzbek and implement those (with examples).  I see you started working on it, and I guess that's what your `TODO` is for :)